### ISSUE TITLE
different result from local run and demo

### ISSUE THREAD

**Issue by xingchen2022** [Stable-X/StableNormal#31]
Thanks for you wonderful work！
I have tried different scripts in reamMe and huggingface, but i can't get the same result like online demo（https://huggingface.co/spaces/Stable-X/StableNormal） 

the online demo result like:
<img width="890" alt="image" src="https://github.com/user-attachments/assets/43806771-375e-4f80-bdaa-11db6e81206a" />

the local results as follows:
torch.hub.load("Stable-X/StableNormal", "StableNormal", trust_repo=True, yoso_version='yoso-normal-v0-3')
![normal_mapv0-3](https://github.com/user-attachments/assets/a9ed9c73-3785-43e4-bf0a-ea9246313bd1)

torch.hub.load("Stable-X/StableNormal", "StableNormal", trust_repo=True, yoso_version='yoso-normal-v1-0')
![normal_mapv1-0](https://github.com/user-attachments/assets/822074a4-4497-48d2-8e45-8e7daed25a22)

torch.hub.load("Stable-X/StableNormal", "StableNormal_turbo", trust_repo=True, yoso_version='yoso-normal-v0-3')
![normal_map_turbov0-3](https://github.com/user-attachments/assets/d3183e4d-27c2-48c4-abb5-fbc40165d451)

torch.hub.load("Stable-X/StableNormal", "StableNormal_turbo", trust_repo=True, yoso_version='yoso-normal-v1-0')
![normal_map_turbov1-0](https://github.com/user-attachments/assets/c24d66ca-86d9-488e-8b42-3477e2f6145b)

torch.hub.load("hugoycj/StableNormal", "StableNormal", trust_repo=True, yoso_version='yoso-normal-v1-5')
![normal_mapv1-5](https://github.com/user-attachments/assets/94b6bc9d-39ce-45a8-beb2-6f371cddde44)

torch.hub.load("hugoycj/StableNormal", "StableNormal_turbo", trust_repo=True, yoso_version='yoso-normal-v1-5')
![normal_map_turbov1_5](https://github.com/user-attachments/assets/18854d80-07ff-494b-ae7d-0f10c05a09c6)

could you give me some hints? thank you in advance！

**Comment by xingchen2022** [comment]
I wonder what's the version of yoso module is used in the online demo? 
Very appreciate if you could give me some advice : ) @hugoycj

**Comment by hugoycj** [comment]
@xingchen2022 Sorry for bothering you! I'm currently occupied with the new release version scheduled for the end of this month and don't have sufficient workload to address the issues in the current release.

The main diff between gradio and torch.hub versions is that torch.hub pads image to square before inference while gradio resizes directly, mainly a resolution difference. We avoid direct resizing as it may cause extra artifacts. For similar online demo performance, use cropping mode like this:
 
```
import torch
from PIL import Image

input_image = Image.open("path/to/your/image.jpg")
predictor = torch.hub.load("Stable-X/StableNormal", "StableNormal", trust_repo=True)
normal_image = predictor(input_image, mode='stable', preprocess='crop')
normal_image.save("output/normal_map.png")
```

It's worth noting that the yoso-normal-v1-5 version is still unstable at present. Hence, we don't recommend using the normal output from this version for supervision. Hope this helps!

**Comment by zxbzhineng** [comment]
@hugoycj  Hello, great work. Excuse me, but I would like to ask you if there is a new stable-normal weight model and when it will be released?

**Comment by hugoycj** [comment]
@zxbzhineng @xingchen2022 Thanks for watching this project!  We're excited to announce a pre-release version of StableNormal_turbo, now available on Hugging Face Spaces: https://huggingface.co/spaces/Stable-X/StableNormal_turbo_beta. For using the new model locally, you can try:
'''
import torch
from PIL import Image

# Load your input image
input_image = Image.open("path/to/your/image.jpg")

# Initialize the model
predictor = torch.hub.load(
    "hugoycj/StableNormal",
    "StableNormal_turbo",
    trust_repo=True,
    yoso_version='yoso-normal-v1-8-1'
)

# Generate normal map
# For images with alpha channel, the background will be automatically masked
# Otherwise, the model will use birefnet for processing
normal_map = predictor(rgba_image, data_type="object")  # This version is only trained with object data, so we recommand using object mode only

# Save the output
normal_map.save("output/normal_map.png")
'''

This is an early release focused on gathering feedback. We're still actively working on improving its stability and robustness.

**Comment by LunaBright** [comment]
> [@zxbzhineng](https://github.com/zxbzhineng) [@xingchen2022](https://github.com/xingchen2022) Thanks for watching this project! We're excited to announce a pre-release version of StableNormal_turbo, now available on Hugging Face Spaces: https://huggingface.co/spaces/Stable-X/StableNormal_turbo_beta. For using the new model locally, you can try:
> 
> ```
> import torch
> from PIL import Image
> 
> # Load your input image
> input_image = Image.open("path/to/your/image.jpg")
> 
> # Initialize the model
> predictor = torch.hub.load(
>     "hugoycj/StableNormal",
>     "StableNormal_turbo",
>     trust_repo=True,
>     yoso_version='yoso-normal-v1-8-1'
> )
> 
> # Generate normal map
> # For images with alpha channel, the background will be automatically masked
> # Otherwise, the model will use birefnet for processing
> normal_map = predictor(rgba_image, data_type="object")  # This version is only trained with object data, so we recommand using object mode only
> 
> # Save the output
> normal_map.save("output/normal_map.png")
> ```
> 
> This is an early release focused on gathering feedback. We're still actively working on improving its stability and robustness.这是早期版本，主要目的是收集反馈。我们仍在积极努力提高其稳定性和健壮性。

Hello, this model doesn't seem to work on the website right now, I get the following error every time I try. Another question is, where exactly can I download the weights file? I Can't seem to find the download address, can you tell me?

<img width="3823" height="1837" alt="Image" src="https://github.com/user-attachments/assets/699f05db-a583-4f15-8406-48034e606466" />
